### PR TITLE
Feature/webinar anon participant

### PR DESCRIPTION
**Pull request by bustardcelly**
Allows for viewing the webinar as an anonymous viewer.

`/join/anon/${token}`

### CODE REVIEW

File: src/components/MainStage/MainStage.tsx [hunk 1]
```diff
@@ -98,11 +99,18 @@ const MainStage = (props: IMainStageProps) => {
 
   const { classes: chatClasses } = useChatStyles()
 
-  if (!mediaStream || !getStreamGuid()) {
+  if (!isAnonymous && (!mediaStream || !getStreamGuid())) {
```
Comment (LourdesBl) [inline src/components/MainStage/MainStage.tsx]: will the Anonymous Audience access to the watchparty as well as to the webinar?

Comment (bustardcelly) [reply src/components/MainStage/MainStage.tsx]: Presumably? Not sure... good question :)

File: src/components/MainStage/MainStageWrapper.tsx [hunk 2]
```diff
@@ -218,7 +255,33 @@ const MainStageWrapper = () => {
...
+      clearTimeout(anonJoinRef.current)
+    }
+    clearTimeout(anonJoinTimeout)
+  }
+  const onAnonymousEntry = () => {
+    stopAnonJoinTimeout()
+    anonJoinTimeout = setTimeout(() => {
+      const { url, request } = getAnonymousSocketUrl(joinToken)
+      join(url, request)
+    }, 2000)
```
Comment (LourdesBl) [inline src/components/MainStage/MainStageWrapper.tsx]: why is this timeout needed here?

Comment (bustardcelly) [reply src/components/MainStage/MainStageWrapper.tsx]: Good catch. It was originally added to try and find where a continual hook re-render was occurring. I believe I found the reason and a proper solution and this was left around.

Will see if removing it still acts as expected.

**Review by LourdesBl — APPROVED** [review]

**Pull request merged**